helix-core/src/syntax.rs

```diff
@@ -1400,6 +1410,95 @@ impl HighlightConfiguration {
 
         self.highlight_indices.store(Arc::new(indices));
     }
+
+    fn injection_pair<'a>(
+        &self,
+        query_match: &QueryMatch<'a, 'a>,
+        source: RopeSlice<'a>,
+    ) -> (Option<InjectionLanguageMarker<'a>>, Option<Node<'a>>) {
+        let mut injection_capture = None;
+        let mut content_node = None;
+
+        for capture in query_match.captures {
+            let index = Some(capture.index);
+            if index == self.injection_language_capture_index {
+                let name = byte_range_to_str(capture.node.byte_range(), source);
+                injection_capture = Some(InjectionLanguageMarker::Name(name));
+            } else if index == self.injection_filename_capture_index {
+                let name = byte_range_to_str(capture.node.byte_range(), source);
+                let path = Path::new(name.as_ref()).to_path_buf();
+                injection_capture = Some(InjectionLanguageMarker::Filename(path.into()));
+            } else if index == self.injection_shebang_capture_index {
+                let node_slice = source.slice(capture.node.byte_range());
+
+                // some languages allow space and newlines before the actual string content
+                // so a shebang could be on either the first or second line
+                let lines = if node_slice.len_lines() > 1 {
+                    node_slice.slice(
+                        // start of first line
+                        node_slice.line_to_byte(0)
+                // end of second line
+                ..node_slice.line_to_byte(2) - 1,
```

Comment:
The formatting looks a bit weird here - did `cargo fmt` do that? It could be worth separating out

```rust
let start = node_slice.line_to_byte(0);
let end = node_slice.line_to_byte(2) - 1;
node_slice.slice(start..end)
```

to avoid it. Or maybe moving the comments to outside of the call to `slice` might do it. I think that may be a bug upstream in `cargo fmt`

Reply:
I went with:
```rust
node_slice.slice({
    let start = node_slice.line_to_byte(0);
    let end = node_slice.line_to_byte(2) - 1;
    start..end
})
```
I do think it is a bug, as the formatter basically just ingored those lines. I could indent or dedent as much as I want, but the formatter wouldn't move them.